src/elements/toniq-dropdown/toniq-dropdown.test.ts

```diff
@@ -0,0 +1,62 @@
+import {assert, expect, fixture} from '@open-wc/testing';
+import {assign, html} from 'element-vir';
+import {queryThroughShadow} from '../../element-testing/query-through-shadow';
+import {ToniqDropdown} from './toniq-dropdown.element';
+
+describe(ToniqDropdown.tagName, () => {
+    const list = [
+        {
+            value: 1,
+            label: 'Option 1',
+        },
+        {
+            value: 2,
+            label: 'Option 2',
+        },
+        {
+            value: '3',
+            label: 'Option 3',
+        },
+    ];
+
+    it('should be registered as a custom element', () => {
+        const newlyCreated = document.createElement(ToniqDropdown.tagName);
+        assert.instanceOf(newlyCreated, ToniqDropdown);
+    });
+
+    it('should correctly set default option', async () => {
+        const rendered = await fixture(
+            html`
+                <${ToniqDropdown} ${assign(ToniqDropdown.props.list, list)} />
+            `,
+        );
+
+        const selectedOption = queryThroughShadow('span.select-selected', rendered);
+        expect(selectedOption).to.have.text(list[0]!.label);
```

Comment:
If you type `list` with `as const` at the end, these not-null-assertions won't be necessary.

Reply:
![Screen Shot 2022-06-07 at 8 07 42 AM](https://user-images.githubusercontent.com/99699764/172269073-80e0132c-dabc-4d20-a7b6-a035c63b7a09.png)

Is this right?

Reply:
![Screen Shot 2022-06-07 at 11 32 40 AM](https://user-images.githubusercontent.com/99699764/172290074-4f425709-54b9-40e4-85ba-bc54f5457e5b.png)

Will have this ff issue

Reply:
Ah! To fix this change the element's input types to the following:
```
        list: [] as Readonly<ToniqDropdownOption[]>,
        select: undefined as undefined | Readonly<ToniqDropdownOption>,
```

Note the added `Readonly`.